p4rt_client/p4rt_client.go

```diff
@@ -571,6 +571,18 @@ func (p *P4RTClient) StreamChannelGet(streamName *string) *P4RTClientStream {
 }
 
 func (p *P4RTClient) streamChannelDestroyInternal(cStream *P4RTClientStream, rErr error) error {
+	// Notify listener
```

Comment:
It is not advisable to do this BEFORE removing the stream from the p.streams map (protected by p.client_mu), as this may lead to race conditions regarding the p.streams map lookup.

Reply:
Ok. That should be fine. The basic idea is to signal the Cond.Wait() in GetArbitration() and GetPacket() after populating the StreamTermErr.

In the present day, if `StreamChannelGetArbitrationResp` returns an error and the error is a StreamTermErr, we still are not sure that the StreamTermErr channel has populated. 

The fix for that would be to populate the StreamTermErr channel before calling cStream.Stop(). In, that scenario we can be pretty sure that if the GetArbitration() has errored because of Stream Termination, then StreamTermErr has populated.

Reply:
```go
func (p *P4RTClient) streamChannelDestroyInternal(cStream *P4RTClientStream, rErr error) error {
	if glog.V(1) {
		glog.Infof("'%s' Cleaning up '%s'", p, cStream)
	}

	// Remove from map
	p.client_mu.Lock()
	if p.streams != nil {
		if _, found := p.streams[cStream.Params.Name]; found {
			delete(p.streams, cStream.Params.Name)
		}
	}
	p.client_mu.Unlock()

	// Notify listener
	streamParams := cStream.Params // Make a copy
	clientParams := p.Params
	// Buffered Channel (will not get stuck unless the receiver is not reading)
	if p.StreamTermErr != nil {
		p.StreamTermErr <- &P4RTStreamTermErr{
			ClientParams: &clientParams,
			StreamParams: &streamParams,
			StreamErr:    rErr,
		}
	}

	cStream.Stop()

	return nil
}
```

Reply:
Yep, i was going to suggest you move the Stop out, great!
Do me a favor (missed that from your last PR), can you please fix the Stop() to NOT have the nested p.stopMU.Lock?
So:
	p.stopMu.Lock()
	p.stop = true
        p.stopMu.Unlock()  <-- move this here

	// Signal any waiting GetArbitration or GetPacket routines.
	p.arb_mu.Lock()
	p.arbCond.Signal()
	p.arb_mu.Unlock()
<-- Add space here for code readability
	p.pkt_mu.Lock()
	p.pktCond.Signal()
	p.pkt_mu.Unlock()

Reply:
Sure!

Reply:
Done!